### PR TITLE
Add intermediate script for complexity

### PR DESCRIPTION
**Pull request by nateglims**
*Description of changes:* Moved complexity to a subscript so we can compensate for it's questionable exit code practices.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tools/ci/complexity.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+exit codes.
+
+USAGE
+}
+
+if [ $# -eq 0 ] ; then
+    usage
+    exit 1
+fi
+
```
Comment (gkwicker) [inline tools/ci/complexity.sh:17]: You could also consider checking to make sure that `complexity` is installed first, e.g.:

```sh
which complexity>/dev/null
if [ $? != 0 ]
then
    echo "complexity is not installed"
    exit 1
fi
```

Comment (nateglims) [reply tools/ci/complexity.sh:18]: done

File: tools/ci/complexity.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+function usage {
+    cat <<USAGE
+Usage: $0 [files...]
+
+A script to run complexity in CI. This is needed to avoid issues with
```
Comment (sarenameas) [inline tools/ci/complexity.sh:7]: This is uncommented

Comment (nateglims) [reply tools/ci/complexity.sh:7]: It's all inside a here doc: https://en.wikipedia.org/wiki/Here_document#Unix_shells

Comment (sarenameas) [reply tools/ci/complexity.sh:7]: thank you

**Pull request merged**